Storm 616 : Storm-jdbc connector.

Parth-Brahmbhatt opened the pull request.

harshach commented:
@Parth-Brahmbhatt  looks like you might have unintended commit as part of this PR.
"STORM-586: TridentKafkaEmitter should catch updateOffsetException."

harshach commented:
@Parth-Brahmbhatt few things to note
1) add external/storm-jdbc to storm/pom.xml under modules
2) I didn't' find any dependency in storm-jdbc/pom.xml for com.google.common and its causing compilation errors  package com.google.common.base does not exist

harshach commented on external/storm-jdbc/README.md:
```diff
@@ -0,0 +1,117 @@
+config.put("jdbc.conf", hikariConfigMap);
+
+JdbcBolt bolt = new JdbcBolt("user_details", jdbcMapper)
+        .withConfigKey("jdbc.conf");
```
is jdbc.conf is a properties file? if so withConfigKey can be renamed to withConfigFile. It will be great if you can add a sample jdbc.conf file to the README.

Parth-Brahmbhatt commented:
* I noticed that and reverted those files to their state in master.
* Added storm-jdbc to storm-core.
* sorry about this, I accidently deleted it with mysql dependency that i added for testing. I have added it back.
* jdbc.conf is just a key in storm config that holds a map for hikariCP configuration. I have added a link to HikariCP configuration document.
